pkg/consensus/retry_watcher.go

```diff
@@ -30,16 +34,29 @@ type retryWatcher struct {
 	getOpts   *etcd.GetOptions
 	watchOpts *etcd.WatcherOptions
 
+	refreshPeriod time.Duration
```

Comment:
optional: seems like you could more succinctly use a `time.Ticker` here, with a non-blocking select below.

In fact, I might just pass in a `<-chan time.Time` to `RetryWatcher` and have the caller own the ticker itself (so that it can properly `Stop` it). Note that a nil channel would be perfectly valid and would never select.

For the test below, I'd just make my own buffered channel and feed it a `time.Time{}` at the right point (which also makes it a 100% deterministic test, which technically the test below isn't since there's no guarantee the 10 minutes period doesn't elapse early).

I generally try to avoid lots of `time.Now()` calls -- they're pretty cheap, but they're still system calls. I don't expect it to be a performance problem here though.